src/components/colorscale/calc.js

```diff
@@ -20,7 +20,7 @@ module.exports = function calc(trace, vals, containerStr, cLetter) {
 
     if(containerStr) {
         container = Lib.nestedProperty(trace, containerStr).get();
-        inputContainer = Lib.nestedProperty(trace._input, containerStr).get();
+        inputContainer = Lib.nestedProperty(trace, containerStr).get();
```

Comment:
What we should do is: mark those attribute with a special flag in the `attributes.js` and handle them separately in `restyle` / `relayout` so that we don't have to mutate `gd.data`. But, that will be for a future PR.